.github/workflows/publish-doc-site.yml

```diff
@@ -0,0 +1,30 @@
+# This is a basic workflow to help you get started with Actions
+
+name: publish doc site
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Build the doc site
+        run: yarn build
+        
+      - name: Publish to gh-pages branch
+        run: gh-pages -d './docs/dist'
```

Comment:
I'm unsure if we need to install gh-pages in the workflow or not - It's set up in the package.json 
https://github.com/CMSgov/mgov-design-system/blob/master/package.json#L59